src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -294,17 +294,10 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
         private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
 
         [TestCaseSource("ZeroTestCasesSource")]
-        public void TestWithZeroTestSourceCasesShouldNotBeRun()
-        {
-            Assert.Fail();
-        }
```

Comment:
I think there is value to both tests. I know this is redundant now, but the knowledge that it is redundant is a detail which is subject to change. Each test does something unique and perpendicular: one ensures the test is not executed, the other ensures that NUnit doesn't fail the test prior to running it due to missing a parameter.